pkg/generated/openapi/def.bzl

```diff
@@ -14,17 +15,17 @@ def openapi_library(name, tags, srcs, openapi_targets=[], vendor_targets=[]):
   )
   go_genrule(
       name = "zz_generated.openapi",
-      srcs = srcs + ["//hack/boilerplate:boilerplate.go.txt"],
+      srcs = srcs + ["//" + openapi_pkg_prefix + "hack/boilerplate:boilerplate.go.txt"],
       outs = ["zz_generated.openapi.go"],
       cmd = " ".join([
         "$(location //vendor/k8s.io/code-generator/cmd/openapi-gen)",
         "--v 1",
         "--logtostderr",
-        "--go-header-file $(location //hack/boilerplate:boilerplate.go.txt)",
+        "--go-header-file $(location //" + openapi_pkg_prefix + "hack/boilerplate:boilerplate.go.txt)",
         "--output-file-base zz_generated.openapi",
-        "--output-package k8s.io/kubernetes/pkg/generated/openapi",
-        "--input-dirs " + ",".join(["k8s.io/kubernetes/" + target for target in openapi_targets] + ["k8s.io/kubernetes/vendor/" + target for target in vendor_targets]),
-        "&& cp pkg/generated/openapi/zz_generated.openapi.go $(location :zz_generated.openapi.go)",
+        "--output-package " + openapi_output_prefix + "k8s.io/kubernetes/pkg/generated/openapi",
```

Comment:
Do you necessarily need this one? Since this is a genrule, you can generate wherever you like and then copy from that location to your final location. It shouldn't matter where this script is generating, as long as you can retrieve the output from it.

Reply:
When ``--output-package`` is not updated to specify the vendored output path, ``openapi-gen`` generates the file outside the repo (i.e. to ``$GOPATH/src/k8s.io/kubernetes/pkg/generated/openapi`` instead of ``$GOPATH/src/bazel-build/k8s.io/federation/vendor/k8s.io/kubernetes/pkg/generated/openapi``).  Yes, the file could be copied from any location, but I think it's preferable to have the output be in the working tree rather than outside it.